src/pages/todo/components/TodoList.js

```diff
@@ -13,18 +14,84 @@ import { LIST_TODOS } from '../actions';
   };
 })
 class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isEditing: false,
+      form: { 
+        title: '' 
+      }
+    }
+  }
+
+  componentDidUpdate() {
+    this.refs.textbox.focus()
+  }
+
+  _validateData(data) {
+    if(data.title == "") {
+      return false
+    }
+    return true
+  }
+
+  _submitForm() {
+    // Handle submit
+    if(this._validateData(this.state.form)) {
+      this.props.dispatch({type: CREATE_TODO.REQUEST, data: this.state.form})
+    }
```

Comment:
Ah, this code does not take care of "async request": you can't handle error nor show loader.

To do this, you should first add the following method in this TodoList:
```
componentWillReceiveProps(nextProps) {
  const {error, fetching} = nextProps[CREATE_TODO.BASE];
  if (!error && !fetching && this.state.editing) {
    this.setState({form: {'title': ''}})
    this.setState({isEditing: false})
  }
  // TODO error handling and show loader (for other issues)
}
```

Then, modify `reduceApi` in common/api.js as follows:
```
export function reduceApi(state, action, requestTypes, onSuccess) {
  switch (action.type) {
    case requestTypes.REQUEST:
      let nextState = Object.assign({}, state);
      nextState[requestTypes.BASE] = {error: null, fetching: true};
      return nextState;
    case requestTypes.FAILURE:
      let nextState = Object.assign({}, state);
      nextState[requestTypes.BASE] = {error: action.error, fetching: false};
      return nextState;
    case requestTypes.SUCCESS:
      let nextState = Object.assign({}, state, onSuccess(action.data));
      nextState[requestTypes.BASE] = {error: null, fetching: false},;
      return nextState;
  }
  return state;
}
```